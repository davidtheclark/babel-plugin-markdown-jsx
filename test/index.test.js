const babel = require('babel-core');
const babelEs2015 = require('babel-preset-es2015');
const babelReact = require('babel-preset-react');
const plugin = require('../');

const transform = code => {
  const module = `
    const React = require('react');
    const Markdown = require('babel-plugin-transform-markdown-in-jsx/component');
    function TestComponent() {
      ${code};
    }`;
  return babel.transform(module, {
    presets: [babelEs2015, babelReact],
    plugins: [plugin]
  }).code;
};

test('basic usage', () => {
  const renderBody = `return (
    <Markdown>
      # Title

      This is **bold.**

      Here is a [link](/some/url).
    </Markdown>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('basic usage inline', () => {
  const renderBody = `return (
    <div>
      Here is <Markdown inline>_interpolated_ markdown</Markdown>
    </div>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('nested jsx', () => {
  const renderBody = `var number = 4;
    return (
      <Markdown>
        Here is a number: <span style={{ fontWeight: 'bold' }}>{number}</span>. Here it is again: {number}.
      </Markdown>
    )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('extra props to element', () => {
  const renderBody = `return (
    <p>
      Here is <Markdown inline style={{ background: '#eee' }}>_interpolated_ markdown</Markdown>
    </p>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('syntax highlighting', () => {
  const renderBody = `return (
    <Markdown>
      Here is a [link](#thing).

      \`\`\`js
      var thing = 'two' === 48;
      var three = 4;
      \`\`\`
    </Markdown>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('nest Markdown components', () => {
  const renderBody = `return (
    <Markdown>
      **markdown**
      <div>
        **no markdown**
      </div>
      <Markdown>
        **markdown**
      </Markdown>
      {/*common comments*/}
    </Markdown>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});
