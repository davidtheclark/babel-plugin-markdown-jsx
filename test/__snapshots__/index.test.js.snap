// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic usage 1`] = `
"'use strict';

var React = require('react');
var Markdown;
function TestComponent() {
  return React.createElement(
    'div',
    null,
    React.createElement(
      'h1',
      null,
      'Title'
    ),
    React.createElement(
      'p',
      null,
      'This is ',
      React.createElement(
        'strong',
        null,
        'bold.'
      )
    ),
    React.createElement(
      'p',
      null,
      'Here is a ',
      React.createElement(
        'a',
        { href: '/some/url' },
        'link'
      ),
      '.'
    )
  );
}"
`;

exports[`basic usage inline 1`] = `
"'use strict';

var React = require('react');
var Markdown;
function TestComponent() {
  return React.createElement(
    'div',
    null,
    'Here is ',
    React.createElement(
      'span',
      null,
      React.createElement(
        'em',
        null,
        'interpolated'
      ),
      ' markdown'
    )
  );
}"
`;

exports[`extra props to element 1`] = `
"'use strict';

var React = require('react');
var Markdown;
function TestComponent() {
  return React.createElement(
    'p',
    null,
    'Here is ',
    React.createElement(
      'span',
      { style: { background: '#eee' } },
      React.createElement(
        'em',
        null,
        'interpolated'
      ),
      ' markdown'
    )
  );
}"
`;

exports[`nest Markdown components 1`] = `
"'use strict';

var React = require('react');
var Markdown;
function TestComponent() {
  return React.createElement(
    'div',
    null,
    React.createElement(
      'p',
      null,
      React.createElement(
        'strong',
        null,
        'markdown'
      )
    ),
    React.createElement(
      'div',
      null,
      '**no markdown**'
    ),
    React.createElement(
      'div',
      null,
      React.createElement(
        'p',
        null,
        React.createElement(
          'strong',
          null,
          'markdown'
        )
      )
    )
  );
}"
`;

exports[`nested jsx 1`] = `
"'use strict';

var React = require('react');
var Markdown;
function TestComponent() {
  var number = 4;
  return React.createElement(
    'div',
    null,
    React.createElement(
      'p',
      null,
      'Here is a number: ',
      React.createElement(
        'span',
        { style: { fontWeight: 'bold' } },
        number
      ),
      '. Here it is again: ',
      number,
      '.'
    )
  );
}"
`;

exports[`syntax highlighting 1`] = `
"'use strict';

var React = require('react');
var Markdown;
function TestComponent() {
  return React.createElement(
    'div',
    null,
    React.createElement(
      'p',
      null,
      'Here is a ',
      React.createElement(
        'a',
        { href: '#thing' },
        'link'
      ),
      '.'
    ),
    React.createElement(
      'pre',
      null,
      React.createElement(
        'code',
        { className: 'language-js' },
        React.createElement(
          'span',
          { className: 'token keyword' },
          'var'
        ),
        ' thing ',
        React.createElement(
          'span',
          { className: 'token operator' },
          '='
        ),
        ' ',
        React.createElement(
          'span',
          { className: 'token string' },
          '\\\\'two\\\\''
        ),
        ' ',
        React.createElement(
          'span',
          { className: 'token operator' },
          '==='
        ),
        ' ',
        React.createElement(
          'span',
          { className: 'token number' },
          '48'
        ),
        React.createElement(
          'span',
          { className: 'token punctuation' },
          ';'
        ),
        \\"\\\\n\\",
        React.createElement(
          'span',
          { className: 'token keyword' },
          'var'
        ),
        ' three ',
        React.createElement(
          'span',
          { className: 'token operator' },
          '='
        ),
        ' ',
        React.createElement(
          'span',
          { className: 'token number' },
          '4'
        ),
        React.createElement(
          'span',
          { className: 'token punctuation' },
          ';'
        ),
        \\"\\\\n\\"
      )
    )
  );
}"
`;
